src/main/webapp/script.js

```diff
@@ -1,18 +1,26 @@
 async function displayCodes() {
-    const response = await fetch('/promo-code');
+    var formInput = document.getElementById('formInput').value;
+    const response = await fetch('/promo-code?formInput=' + formInput);
     const codes = await response.json();
     const tableEl = document.getElementById('promoCodeTable');
     tableEl.innerHTML =
                 `<tr>
                     <th>VideoURL</th>
                     <th>PROMO CODE</th>
                 </tr>`;
-    const numOfCodes = Object.keys(codes).length;
-    for (i = 0; i < numOfCodes; i++) {
+    if (codes == 0) {
         var row = tableEl.insertRow(-1);
-        var videoUrl = row.insertCell(0);
-        var promoCode = row.insertCell(1);
-        videoUrl.innerHTML = '<a href="' + codes[i].videoUrl +'">' + codes[i].videoUrl + '</a>'
-        promoCode.innerHTML = codes[i].promoCode;
+        var noCodes = row.insertCell(0);
+        noCodes.innerHTML = '<p>Sorry! There are no codes with this Id.</p>'
+    } else {
+        const numOfCodes = Object.keys(codes).length;
+        for (i = 0; i < numOfCodes; i++) {
+            var row = tableEl.insertRow(-1);
```

Comment:
Variable `row` is already defined in line 12. Either remove or consider using `let` instead, to make the variable block scoped and not function scoped.